linera-service/src/client.rs

```diff
@@ -0,0 +1,974 @@
+const CLIENT_SERVICE_ENV: &str = "INTEGRATION_TEST_CLIENT_SERVICE_PARAMS";
+
+#[derive(Copy, Clone)]
+pub enum Network {
+    Grpc,
+    Simple,
+}
+
+impl Network {
+    fn internal(&self) -> &'static str {
+        match self {
+            Network::Grpc => "\"Grpc\"",
+            Network::Simple => "{ Simple = \"Tcp\" }",
+        }
+    }
+
+    fn external(&self) -> &'static str {
+        match self {
+            Network::Grpc => "\"Grpc\"",
+            Network::Simple => "{ Simple = \"Tcp\" }",
+        }
+    }
+
+    fn external_short(&self) -> &'static str {
+        match self {
+            Network::Grpc => "grpc",
+            Network::Simple => "tcp",
+        }
+    }
+}
+
+#[allow(unused_mut)]
+fn detect_current_features() -> Vec<&'static str> {
+    let mut features = vec![];
+    #[cfg(benchmark)]
+    {
+        features.push("benchmark");
+    }
+    #[cfg(wasmer)]
+    {
+        features.push("wasmer");
+    }
+    #[cfg(wasmtime)]
+    {
+        features.push("wasmtime");
+    }
+    #[cfg(rocksdb)]
+    {
+        features.push("rocksdb");
+    }
+    #[cfg(aws)]
+    {
+        features.push("aws");
+    }
+    features
+}
+
+pub async fn cargo_force_build_binary(name: &'static str) -> Result<PathBuf> {
+    let mut build_command = Command::new("cargo");
+    build_command.arg("build");
+    let is_release = if let Ok(var) = env::var(CARGO_ENV) {
+        let extra_args = var.split_whitespace();
+        build_command.args(extra_args.clone());
+        let extra_args: HashSet<_> = extra_args.into_iter().map(str::trim).collect();
+        extra_args.contains("-r") || extra_args.contains("--release")
+    } else {
+        false
+    };
+    // Use the same features as the current environment so that we don't rebuild as often.
+    let features = detect_current_features();
+    if !features.is_empty() {
+        build_command
+            .arg("--no-default-features")
+            .arg("--features")
+            .args(features);
+    }
+    build_command.args(["--bin", name]);
+    info!("Running compiler: {:?}", build_command);
+    assert!(build_command.spawn()?.wait().await?.success());
+    if is_release {
+        Ok(env::current_dir()?
+            .join("../target/release")
+            .join(name)
+            .canonicalize()?)
+    } else {
+        Ok(env::current_dir()?
+            .join("../target/debug")
+            .join(name)
+            .canonicalize()?)
+    }
+}
+
+pub async fn cargo_build_binary(name: &'static str) -> Result<PathBuf> {
+    static COMPILED_BINARIES: OnceCell<Mutex<HashMap<&'static str, PathBuf>>> = OnceCell::new();
+    let mut binaries = COMPILED_BINARIES.get_or_init(Default::default).lock().await;
+    match binaries.get(name) {
+        Some(path) => Ok(path.clone()),
+        None => {
+            let path = cargo_force_build_binary(name).await?;
+            binaries.insert(name, path.clone());
+            Ok(path)
+        }
+    }
+}
+
+pub struct Client {
```

Comment:
We use the word `Client` in lots of contexts. :confused: 
Maybe `ClientWallet` or `ClientDirs` or something?

Reply:
And is it even necessary for the `linera` binary to build itself and call itself?
It makes sense to build and call `server`, but `linera` could just do the work itself, instead of calling another process?

Reply:
No this is a good point. In `cfg(test)` it should, otherwise it should look for the binaries in `.cargo`.

Reply:
I mean in `linera net up`, is it even necessary to call

```rust
client1.create_genesis_config().await?;
client2.wallet_init(&[]).await?;
```

?
Couldn't we just create the genesis config and initialize the wallet directly, without using a different binary or even _process_ than the one that's currently running?